sphinx_gallery/gen_gallery.py

```diff
@@ -738,25 +741,25 @@ def _sec_to_readable(t):
     return t
 
 
-def cost_name_key(cost_name):
+def _cost_key(cost):
     """Cost (time_elapsed, memory_used) sorting function."""
-    cost, name = cost_name
     # sort by descending computation time, descending memory, alphabetical name
-    return (-cost[0], -cost[1], name)
+    return (-cost["t"], -cost["mem"], cost["src_file"])
 
 
-def _format_for_writing(costs, path, kind="rst"):
+def _format_for_writing(costs, *, src_dir, kind="rst"):
     """Provide formatted computation summary text.
 
     Parameters
     ----------
     costs: List[Tuple[Tuple[float], str]]
         List of tuples of computation costs and absolute paths to example, of format:
         ((time_elapsed, memory_used), example_path).
-    path: str
-        Source directory.
-    kind: 'rst' or 'console', default='rst'
-        Format for printing to 'console' or for writing `sg_execution_times.rst' ('rst')
+    src_dir : pathlib.Path
+        The Sphinx source directory.
+    kind: 'rst', 'rst-full' or 'console', default='rst'
+        Format for printing to 'console' or for writing `sg_execution_times.rst' ('rst'
+        and 'rst-full').
```

Comment:
Could we add a comment about 'rst' = single gallery and 'rst-full' (all galleries)?